La'Tonia/library_challenge.py

```diff
@@ -0,0 +1,254 @@
+    },
+    {
+        'title': 'ON TYRANNY',
+        'author': 'Timothy Snyder',
+        'description': 'Twenty lessons from the 20th century about the course of tyranny.'
+    },
+    {
+        'title': 'THE ROAD TO UNFREEDOM',
+        'author': 'Timothy Snyder',
+        'description': 'Snyder explores Russian attempts to influence Western democracies and the influence of philosopher Ivan Ilyin on Russian President Vladimir Putin and the Russian Federation in general.'
+    }
+]
+# 2.0
+print('EXERCISE 2.0: DATA STRUCTURE OF BOOKS W/DETAILS')
+# Describe the structure of the data in books_with_details. What types of data are nested within others? How do you know?
+# books_with_details is a list with nested dictionaries
+# ran type() to output datatype of assignment to books_with_details followed by type() to output the first object in the list to confirm datatype of dictionary, which is explicitly the same datatype of all the nested dictionaries in the list assigned to books_with_details
+print(type(books_with_details))
+print(type(books_with_details[0]))
+print()
+
+pprint(books_with_details)
+print()
+
+
+# 2.1
+print('EXERCISE 2.1: CREATING COUNT BOOKS FN')
+# Create a function 'count_books' that returns the number of books in the books_with_details list
+# Parameters: Not needed for this function
+# Return: number of books (integer)
+# use of basic defined function syntax with return to store value to be outputted and len to output the number of items in the variable being passed as a second-level parameter
+def count_books():
+    return len(books_with_details)
+
+# 2.2
+print('EXERCISE 2.2: INVOKING COUNT BOOKS FN')
+# Check the number of books available in the books list using the count_books function
+# basic code structure to invoke the fn count_books
+print(count_books())
+print()
+
+# 2.3
+print('EXERCISE 2.3: CREATING SEARCH BY AUTHOR FN')
+# Create a function 'search_by_author' that returns the titles of books by an author
+# Parameters - author (string)
+# Return - author's books (list of strings)
+# # Hint - You will need a for loop, if statement, .append() for this solution!
+def search_by_author(lst4):
+    author_titles = []
+    for author in books_with_details:
+        if lst4 == author['author']:
+            author_titles.append(author['title'])
+        # if search_by_author != author:
+        # if search_by_author != books_with_details[author]['author']:
+            # search_by_author == 'Author Unknown'
+            # print('Author Unknown')
+        else:
+            print('Author Unknown')
+    return author_titles
+   
+print(search_by_author('Resmaa Menakem'))
+print()
+
+
+def search_by_author(str):
+    author_titles = []
+    for i in books_with_details:
+        if str == i['author']:
+            author_titles.append(i['title'])
+            #print(title)
+        else:
```

Comment:
Good use of else, helpful to have, so you wont have an error code if someone enters wrong spelling or something